testutil/host.go

```diff
@@ -0,0 +1,395 @@
+package testutil
+
+import (
+	"crypto/ed25519"
+	"errors"
+	"net"
+	"sync"
+	"testing"
+
+	proto4 "go.sia.tech/core/rhp/v4"
+	"go.sia.tech/core/types"
+	"go.sia.tech/coreutils/chain"
+	rhp4 "go.sia.tech/coreutils/rhp/v4"
```

Comment:
I've been writing this as `rhpv2`/`rhpv3` elsewhere; not a big deal, but we should be consistent one way or the other.

Reply:
`hostd` is all `rhp2` and `rhp3` and `renterd` is a mix of both 😓

Reply:
rip